src/nvidia.cc

```diff
@@ -376,30 +374,40 @@ class nvidia_display_setting
   virtual void lua_setter(lua::state &l, bool init);
   virtual void cleanup(lua::state &l);
 
+  std::string nvdisplay;
+
  public:
   nvidia_display_setting() : Base("nvidia_display", std::string(), false) {}
+  virtual Display *get_nvdisplay();
 };
 
 void nvidia_display_setting::lua_setter(lua::state &l, bool init) {
   lua::stack_sentry s(l, -2);
 
   Base::lua_setter(l, init);
 
-  std::string str = do_convert(l, -1).first;
-  if (!str.empty()) {
-    nvdisplay = XOpenDisplay(str.c_str());
-    if (nvdisplay == nullptr) {
-      CRIT_ERR(nullptr, NULL, "can't open nvidia display: %s",
-               XDisplayName(str.c_str()));
-    }
-  }
+  nvdisplay = do_convert(l, -1).first;
 
   ++s;
 }  // namespace
 
+Display *nvidia_display_setting::get_nvdisplay() {
+  if (!nvdisplay.empty()) {
+    Display *nvd = XOpenDisplay(nvdisplay.c_str());
```

Comment:
I'm not seeing the corresponding calls to `XCloseDisplay` to release the resources from `XOpenDisplay`.

Alternatively, you could wrap the pointer in a small struct or class and use RAII to call `XCloseDisplay` in the destructor.

Reply:
Yeah, it was bad I forgot about releasing (I made incorrect assumptions about the XDisplay without checking specs). I updated the code with smart pointer to handle this now.